flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/table/catalog/client/PulsarCatalogClient.java

```diff
@@ -0,0 +1,406 @@
+            throws PulsarAdminException, DatabaseNotExistException {
+        if (configuration.isInternalDatabase(name)) {
+            throw new CatalogException(name + " is an internal database, you can't get its tables");
+        }
+
+        String databaseName = configuration.databaseName(name);
+
+        // We only support the persistent topics.
+        ListNamespaceTopicsOptions options =
+                ListNamespaceTopicsOptions.builder()
+                        .includeSystemTopic(false)
+                        .mode(Mode.PERSISTENT)
+                        .build();
+        try {
+            List<String> topics = admin.namespaces().getTopics(databaseName, options);
+            Map<String, String> nameMapping =
+                    nameMapper.listMapping(databaseName).entrySet().stream()
+                            .collect(toMap(Map.Entry::getValue, Map.Entry::getKey));
+
+            // Convert the topic name into table name.
+            List<String> tables = new ArrayList<>(topics.size());
+            for (String topic : topics) {
+                String table = TopicName.get(topic).getLocalName();
+                if (!configuration.isInternalTable(databaseName, table)) {
+                    tables.add(nameMapping.getOrDefault(table, table));
+                }
+            }
+
+            return tables;
+        } catch (NotFoundException e) {
+            throw new DatabaseNotExistException(catalogName, name, e);
+        }
+    }
+
+    public boolean tableExists(ObjectPath tablePath) throws PulsarAdminException {
+        String databaseName = configuration.databaseName(tablePath.getDatabaseName());
+        String tableName = nameMapper.getMapping(databaseName, tablePath.getObjectName());
+        try {
+            String table = TopicName.get(tableName).getLocalName();
+            if (configuration.isInternalTable(databaseName, table)) {
+                return false;
+            }
+
+            admin.topics().getPartitionedTopicMetadata(tableName);
+            return true;
+        } catch (NotFoundException e) {
+            return false;
+        }
+    }
+
+    public void dropTable(ObjectPath tablePath, boolean ignoreIfNotExists)
+            throws PulsarAdminException, TableNotExistException {
+        if (!tableExists(tablePath)) {
+            if (ignoreIfNotExists) {
+                return;
+            } else {
+                throw new TableNotExistException(catalogName, tablePath);
+            }
+        }
+
+        String databaseName = configuration.databaseName(tablePath.getDatabaseName());
+        String tableName = nameMapper.dropMapping(databaseName, tablePath.getObjectName());
+        String table = TopicName.get(tableName).getLocalName();
+        if (configuration.isInternalTable(databaseName, table)) {
+            throw new CatalogException(
+                    "This is a internal table " + tablePath.getObjectName() + " in Pulsar catalog");
+        }
```

Comment:
Maybe this is not needed as `tableExists ` already ensure that this is not a internal database?